### PR TITLE
generating LLC client

### PR DESCRIPTION
**Pull request by mshaban-msft**

### CODE REVIEW

File: sdk/cognitivelanguage/azure-ai-language-questionanswering/azure/ai/language/questionanswering/projects/_question_answering_projects_client.py [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+        # type: (...) -> None
+        _endpoint = '{Endpoint}/language'
+        self._config = QuestionAnsweringProjectsClientConfiguration(endpoint=endpoint, credential=credential, **kwargs)
+        self._client = PipelineClient(base_url=_endpoint, config=self._config, **kwargs)
+
+        client_models = {k: v for k, v in models.__dict__.items() if isinstance(v, type)}
+        self._serialize = Serializer(client_models)
+        self._deserialize = Deserializer(client_models)
+        self._serialize.client_side_validation = False
+        self.question_answering_projects = QuestionAnsweringProjectsOperations(self._client, self._config, self._serialize, self._deserialize)
```
Comment (annatisch) [inline sdk/cognitivelanguage/azure-ai-language-questionanswering/azure/ai/language/questionanswering/projects/_question_answering_projects_client.py:59]: Can we regenerate this with the operation group name as just `projects`?

File: sdk/cognitivelanguage/azure-ai-language-questionanswering/azure/ai/language/questionanswering/projects/aio/__init__.py [hunk 2]
```diff
@@ -0,0 +1,16 @@
+# coding=utf-8
+# --------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# Code generated by Microsoft (R) AutoRest Code Generator.
+# Changes may cause incorrect behavior and will be lost if the code is regenerated.
+# --------------------------------------------------------------------------
```
Comment (annatisch) [inline sdk/cognitivelanguage/azure-ai-language-questionanswering/azure/ai/language/questionanswering/projects/aio/__init__.py:7]: Maybe for the next preview we should look into whether this namespace should be `.projects.aio` or `.aio.projects`.
But we may remove this `projects` namespace altogether, so we can have that discussion later.

File: sdk/cognitivelanguage/azure-ai-language-questionanswering/azure/ai/language/questionanswering/projects/models/_question_answering_projects_client_enums.py [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+# Code generated by Microsoft (R) AutoRest Code Generator.
+# Changes may cause incorrect behavior and will be lost if the code is regenerated.
+# --------------------------------------------------------------------------
+
+from enum import Enum
+from six import with_metaclass
+from azure.core import CaseInsensitiveEnumMeta
+
+
+class Enum4(with_metaclass(CaseInsensitiveEnumMeta, str, Enum)):
```
Comment (annatisch) [inline sdk/cognitivelanguage/azure-ai-language-questionanswering/azure/ai/language/questionanswering/projects/models/_question_answering_projects_client_enums.py:14]: This needs a client name

File: sdk/cognitivelanguage/azure-ai-language-questionanswering/azure/ai/language/questionanswering/projects/aio/operations/_operations.py [hunk 4]
```diff
@@ -0,0 +1,1815 @@
...
+        self._client = client
+        self._serialize = serializer
+        self._deserialize = deserializer
+        self._config = config
+
+    @distributed_trace
+    def list_projects(
+        self,
+        *,
+        top: Optional[int] = None,
```
Comment (annatisch) [inline sdk/cognitivelanguage/azure-ai-language-questionanswering/azure/ai/language/questionanswering/projects/aio/operations/_operations.py:54]: What's the functional difference between `top` and `maxpagesize`? Will `top` act as a limit for the items in _all_ pages?

File: sdk/cognitivelanguage/azure-ai-language-questionanswering/azure/ai/language/questionanswering/projects/aio/operations/_operations.py [hunk 5]
```diff
@@ -0,0 +1,1815 @@
...
+            return pipeline_response
+
+
+        return AsyncItemPaged(
+            get_next, extract_data
+        )
+    list_projects.metadata = {'url': '/query-knowledgebases/projects'}  # type: ignore
+
+    @distributed_trace_async
+    async def get_project_details(
```
Comment (annatisch) [inline sdk/cognitivelanguage/azure-ai-language-questionanswering/azure/ai/language/questionanswering/projects/aio/operations/_operations.py:146]: Maybe this should be `get_project` to match `list_projects`?

File: sdk/cognitivelanguage/azure-ai-language-questionanswering/azure/ai/language/questionanswering/projects/aio/operations/_operations.py [hunk 6]
```diff
@@ -0,0 +1,1815 @@
...
+        return deserialized
+
+    get_project_details.metadata = {'url': '/query-knowledgebases/projects/{projectName}'}  # type: ignore
+
+
+    @distributed_trace_async
+    async def create_project(
+        self,
+        project_name: str,
+        body: "_models.CreateProjectOptions",
```
Comment (annatisch) [inline sdk/cognitivelanguage/azure-ai-language-questionanswering/azure/ai/language/questionanswering/projects/aio/operations/_operations.py:205]: Can we rename the `body` parameter to `options`.

File: sdk/cognitivelanguage/azure-ai-language-questionanswering/azure/ai/language/questionanswering/projects/aio/operations/_operations.py [hunk 7]
```diff
@@ -0,0 +1,1815 @@
...
+        if cls:
+            return cls(pipeline_response, deserialized, {})
+
+        return deserialized
+
+    create_project.metadata = {'url': '/query-knowledgebases/projects/{projectName}'}  # type: ignore
+
+
+    @distributed_trace_async
+    async def delete_project(
```
Comment (annatisch) [inline sdk/cognitivelanguage/azure-ai-language-questionanswering/azure/ai/language/questionanswering/projects/aio/operations/_operations.py:269]: Is this the function that needs to be regenerated as an LRO? Has that swagger fix been merged?

File: sdk/cognitivelanguage/azure-ai-language-questionanswering/azure/ai/language/questionanswering/projects/aio/operations/_operations.py [hunk 8]
```diff
@@ -0,0 +1,1815 @@
...
+
+
+        if cls:
+            return cls(pipeline_response, None, response_headers)
+
+    delete_project.metadata = {'url': '/query-knowledgebases/projects/{projectName}'}  # type: ignore
+
+
+    @distributed_trace_async
+    async def get_delete_status(
```
Comment (annatisch) [inline sdk/cognitivelanguage/azure-ai-language-questionanswering/azure/ai/language/questionanswering/projects/aio/operations/_operations.py:325]: We can skip this API from the generated SDK. We don't usually expose this.

File: sdk/cognitivelanguage/azure-ai-language-questionanswering/azure/ai/language/questionanswering/projects/aio/operations/_operations.py [hunk 9]
```diff
@@ -0,0 +1,1815 @@
...
+
+    @distributed_trace_async
+    async def begin_export(
+        self,
+        project_name: str,
+        *,
+        format: Optional[Union[str, "_models.Format"]] = "json",
+        asset_kind: Optional[Union[str, "_models.Enum4"]] = None,
+        **kwargs: Any
+    ) -> AsyncLROPoller[None]:
```
Comment (annatisch) [inline sdk/cognitivelanguage/azure-ai-language-questionanswering/azure/ai/language/questionanswering/projects/aio/operations/_operations.py:434]: The result of this LRO is `None`, so where is the project exported to?

File: sdk/cognitivelanguage/azure-ai-language-questionanswering/azure/ai/language/questionanswering/projects/aio/operations/_operations.py [hunk 10]
```diff
@@ -0,0 +1,1815 @@
...
+                client=self._client,
+                deserialization_callback=get_long_running_output
+            )
+        else:
+            return AsyncLROPoller(self._client, raw_result, get_long_running_output, polling_method)
+
+    begin_export.metadata = {'url': '/query-knowledgebases/projects/{projectName}/:export'}  # type: ignore
+
+    @distributed_trace_async
+    async def get_export_status(
```
Comment (annatisch) [inline sdk/cognitivelanguage/azure-ai-language-questionanswering/azure/ai/language/questionanswering/projects/aio/operations/_operations.py:503]: We can skip this API from the generated SDK. We don't usually expose this.

File: sdk/cognitivelanguage/azure-ai-language-questionanswering/azure/ai/language/questionanswering/projects/aio/operations/_operations.py [hunk 11]
```diff
@@ -0,0 +1,1815 @@
...
+            return cls(pipeline_response, None, response_headers)
+
+    _import_method_initial.metadata = {'url': '/query-knowledgebases/projects/{projectName}/:import'}  # type: ignore
+
+
+    @distributed_trace_async
+    async def begin_import_method(
+        self,
+        project_name: str,
+        body: Optional["_models.ImportJobOptions"] = None,
```
Comment (annatisch) [inline sdk/cognitivelanguage/azure-ai-language-questionanswering/azure/ai/language/questionanswering/projects/aio/operations/_operations.py:620]: Let's rename `body` to `options`

File: sdk/cognitivelanguage/azure-ai-language-questionanswering/azure/ai/language/questionanswering/projects/aio/operations/_operations.py [hunk 12]
```diff
@@ -0,0 +1,1815 @@
...
+                client=self._client,
+                deserialization_callback=get_long_running_output
+            )
+        else:
+            return AsyncLROPoller(self._client, raw_result, get_long_running_output, polling_method)
+
+    begin_import_method.metadata = {'url': '/query-knowledgebases/projects/{projectName}/:import'}  # type: ignore
+
+    @distributed_trace_async
+    async def get_import_status(
```
Comment (annatisch) [inline sdk/cognitivelanguage/azure-ai-language-questionanswering/azure/ai/language/questionanswering/projects/aio/operations/_operations.py:699]: We can skip this API from the generated SDK. We don't usually expose this.

File: sdk/cognitivelanguage/azure-ai-language-questionanswering/azure/ai/language/questionanswering/projects/aio/operations/_operations.py [hunk 13]
```diff
@@ -0,0 +1,1815 @@
...
+                client=self._client,
+                deserialization_callback=get_long_running_output
+            )
+        else:
+            return AsyncLROPoller(self._client, raw_result, get_long_running_output, polling_method)
+
+    begin_deploy_project.metadata = {'url': '/query-knowledgebases/projects/{projectName}/deployments/{deploymentName}'}  # type: ignore
+
+    @distributed_trace_async
+    async def get_deploy_status(
```
Comment (annatisch) [inline sdk/cognitivelanguage/azure-ai-language-questionanswering/azure/ai/language/questionanswering/projects/aio/operations/_operations.py:873]: We can skip this API from the generated SDK. We don't usually expose this.

File: sdk/cognitivelanguage/azure-ai-language-questionanswering/azure/ai/language/questionanswering/projects/aio/operations/_operations.py [hunk 14]
```diff
@@ -0,0 +1,1815 @@
...
+    @distributed_trace
+    def get_synonyms(
+        self,
+        project_name: str,
+        *,
+        top: Optional[int] = None,
+        skip: Optional[int] = None,
+        maxpagesize: Optional[int] = None,
+        **kwargs: Any
+    ) -> AsyncIterable["_models.SynonymAssets"]:
```
Comment (annatisch) [inline sdk/cognitivelanguage/azure-ai-language-questionanswering/azure/ai/language/questionanswering/projects/aio/operations/_operations.py:1045]: This is returning a pageable, so should be named `list_synonyms`

File: sdk/cognitivelanguage/azure-ai-language-questionanswering/azure/ai/language/questionanswering/projects/aio/operations/_operations.py [hunk 15]
```diff
@@ -0,0 +1,1815 @@
...
+        return AsyncItemPaged(
+            get_next, extract_data
+        )
+    get_synonyms.metadata = {'url': '/query-knowledgebases/projects/{projectName}/synonyms'}  # type: ignore
+
+    @distributed_trace_async
+    async def update_synonyms(
+        self,
+        project_name: str,
+        body: "_models.SynonymAssets",
```
Comment (annatisch) [inline sdk/cognitivelanguage/azure-ai-language-questionanswering/azure/ai/language/questionanswering/projects/aio/operations/_operations.py:1140]: `body` -> `synonyms`?

File: sdk/cognitivelanguage/azure-ai-language-questionanswering/azure/ai/language/questionanswering/projects/aio/operations/_operations.py [hunk 16]
```diff
@@ -0,0 +1,1815 @@
...
+    @distributed_trace
+    def get_sources(
+        self,
+        project_name: str,
+        *,
+        top: Optional[int] = None,
+        skip: Optional[int] = None,
+        maxpagesize: Optional[int] = None,
+        **kwargs: Any
+    ) -> AsyncIterable["_models.QnaSources"]:
```
Comment (annatisch) [inline sdk/cognitivelanguage/azure-ai-language-questionanswering/azure/ai/language/questionanswering/projects/aio/operations/_operations.py:1204]: Also looks like it should be a `list_` operation.

File: sdk/cognitivelanguage/azure-ai-language-questionanswering/azure/ai/language/questionanswering/projects/aio/operations/_operations.py [hunk 17]
```diff
@@ -0,0 +1,1815 @@
...
+            return cls(pipeline_response, None, response_headers)
+
+    _update_sources_initial.metadata = {'url': '/query-knowledgebases/projects/{projectName}/sources'}  # type: ignore
+
+
+    @distributed_trace_async
+    async def begin_update_sources(
+        self,
+        project_name: str,
+        body: List["_models.UpdateSourceRecord"],
```
Comment (annatisch) [inline sdk/cognitivelanguage/azure-ai-language-questionanswering/azure/ai/language/questionanswering/projects/aio/operations/_operations.py:1345]: `body` -> `sources`?

File: sdk/cognitivelanguage/azure-ai-language-questionanswering/azure/ai/language/questionanswering/projects/aio/operations/_operations.py [hunk 18]
```diff
@@ -0,0 +1,1815 @@
...
+                client=self._client,
+                deserialization_callback=get_long_running_output
+            )
+        else:
+            return AsyncLROPoller(self._client, raw_result, get_long_running_output, polling_method)
+
+    begin_update_sources.metadata = {'url': '/query-knowledgebases/projects/{projectName}/sources'}  # type: ignore
+
+    @distributed_trace_async
+    async def get_update_sources_status(
```
Comment (annatisch) [inline sdk/cognitivelanguage/azure-ai-language-questionanswering/azure/ai/language/questionanswering/projects/aio/operations/_operations.py:1415]: Let's not expose this one either.

File: sdk/cognitivelanguage/azure-ai-language-questionanswering/azure/ai/language/questionanswering/projects/aio/operations/_operations.py [hunk 19]
```diff
@@ -0,0 +1,1815 @@
...
+    def get_qnas(
+        self,
+        project_name: str,
+        *,
+        source: Optional[str] = None,
+        top: Optional[int] = None,
+        skip: Optional[int] = None,
+        maxpagesize: Optional[int] = None,
+        **kwargs: Any
+    ) -> AsyncIterable["_models.QnaAssets"]:
```
Comment (annatisch) [inline sdk/cognitivelanguage/azure-ai-language-questionanswering/azure/ai/language/questionanswering/projects/aio/operations/_operations.py:1484]: Also a `list_`

File: sdk/cognitivelanguage/azure-ai-language-questionanswering/azure/ai/language/questionanswering/projects/aio/operations/_operations.py [hunk 20]
```diff
@@ -0,0 +1,1815 @@
...
+            return cls(pipeline_response, None, response_headers)
+
+    _update_qnas_initial.metadata = {'url': '/query-knowledgebases/projects/{projectName}/qnas'}  # type: ignore
+
+
+    @distributed_trace_async
+    async def begin_update_qnas(
+        self,
+        project_name: str,
+        body: List["_models.UpdateQnaRecord"],
```
Comment (annatisch) [inline sdk/cognitivelanguage/azure-ai-language-questionanswering/azure/ai/language/questionanswering/projects/aio/operations/_operations.py:1629]: `body` -> `qnas`

File: sdk/cognitivelanguage/azure-ai-language-questionanswering/azure/ai/language/questionanswering/projects/aio/operations/_operations.py [hunk 21]
```diff
@@ -0,0 +1,1815 @@
...
+                client=self._client,
+                deserialization_callback=get_long_running_output
+            )
+        else:
+            return AsyncLROPoller(self._client, raw_result, get_long_running_output, polling_method)
+
+    begin_update_qnas.metadata = {'url': '/query-knowledgebases/projects/{projectName}/qnas'}  # type: ignore
+
+    @distributed_trace_async
+    async def get_update_qnas_status(
```
Comment (annatisch) [inline sdk/cognitivelanguage/azure-ai-language-questionanswering/azure/ai/language/questionanswering/projects/aio/operations/_operations.py:1699]: Hide this one too.

File: sdk/cognitivelanguage/azure-ai-language-questionanswering/azure/ai/language/questionanswering/projects/aio/operations/_operations.py [hunk 22]
```diff
@@ -0,0 +1,1815 @@
...
+        return deserialized
+
+    get_update_qnas_status.metadata = {'url': '/query-knowledgebases/projects/{projectName}/qnas/jobs/{jobId}'}  # type: ignore
+
+
+    @distributed_trace_async
+    async def add_feedback(
+        self,
+        project_name: str,
+        body: "_models.ActiveLearningFeedback",
```
Comment (annatisch) [inline sdk/cognitivelanguage/azure-ai-language-questionanswering/azure/ai/language/questionanswering/projects/aio/operations/_operations.py:1762]: `body` -> `feedback`

File: sdk/cognitivelanguage/azure-ai-language-questionanswering/azure/ai/language/questionanswering/projects/models/_models_py3.py [hunk 23]
```diff
@@ -0,0 +1,1769 @@
...
+    ):
+        """
+        :keyword errors:
+        :paramtype errors: list[~azure.ai.language.questionanswering.projects.models.Error]
+        """
+        super(Errors, self).__init__(**kwargs)
+        self.errors = errors
+
+
+class ExportJobResultUrl(msrest.serialization.Model):
```
Comment (annatisch) [inline sdk/cognitivelanguage/azure-ai-language-questionanswering/azure/ai/language/questionanswering/projects/models/_models_py3.py:253]: It looks like this should be the result of the `export` LRO poller?

File: sdk/cognitivelanguage/azure-ai-language-questionanswering/azure/ai/language/questionanswering/projects/models/_models_py3.py [hunk 24]
```diff
@@ -0,0 +1,1769 @@
...
+import datetime
+from typing import Dict, List, Optional, Union
+
+from azure.core.exceptions import HttpResponseError
+import msrest.serialization
+
+from ._question_answering_projects_client_enums import *
+
+
+class ActiveLearningFeedback(msrest.serialization.Model):
```
Comment (annatisch) [inline sdk/cognitivelanguage/azure-ai-language-questionanswering/azure/ai/language/questionanswering/projects/models/_models_py3.py:18]: I'm inclined to think we actually try generating this without models. I think that will be cleaner given that we're going with an LLC approach to this side of the SDK.

**Review by annatisch — CHANGES_REQUESTED** [review]

File: sdk/cognitivelanguage/azure-ai-language-questionanswering/azure/ai/language/questionanswering/projects/_question_answering_projects_client.py (hunk 1 above)
Comment (annatisch) [reply sdk/cognitivelanguage/azure-ai-language-questionanswering/azure/ai/language/questionanswering/projects/_question_answering_projects_client.py:59]: Actually all of the operation names already have nouns in them - I think we can drop the operation group entirely.

File: sdk/cognitivelanguage/azure-ai-language-questionanswering/azure/ai/language/questionanswering/projects/aio/operations/_operations.py (hunk 4 above)
Comment (mshaban-msft) [reply sdk/cognitivelanguage/azure-ai-language-questionanswering/azure/ai/language/questionanswering/projects/aio/operations/_operations.py]: `top` limits the total number of results returned 
and `maxpagesize` is the number of results per page

File: sdk/cognitivelanguage/azure-ai-language-questionanswering/azure/ai/language/questionanswering/projects/aio/operations/_operations.py (hunk 7 above)
Comment (mshaban-msft) [reply sdk/cognitivelanguage/azure-ai-language-questionanswering/azure/ai/language/questionanswering/projects/aio/operations/_operations.py]: all functions that has LROs need to be re-generated
delete, export, update knowledge source, ..
and yes the new swagger update is released (but not merged)
and i did the regen

File: sdk/cognitivelanguage/azure-ai-language-questionanswering/azure/ai/language/questionanswering/projects/aio/operations/_operations.py (hunk 11 above)
Comment (mshaban-msft) [reply sdk/cognitivelanguage/azure-ai-language-questionanswering/azure/ai/language/questionanswering/projects/aio/operations/_operations.py]: you made that fix

File: sdk/cognitivelanguage/azure-ai-language-questionanswering/azure/ai/language/questionanswering/projects/aio/operations/_operations.py (hunk 13 above)
Comment (mshaban-msft) [reply sdk/cognitivelanguage/azure-ai-language-questionanswering/azure/ai/language/questionanswering/projects/aio/operations/_operations.py]: you already did that

File: sdk/cognitivelanguage/azure-ai-language-questionanswering/azure/ai/language/questionanswering/projects/models/_models_py3.py (hunk 23 above)
Comment (mshaban-msft) [reply sdk/cognitivelanguage/azure-ai-language-questionanswering/azure/ai/language/questionanswering/projects/models/_models_py3.py]: yes!

File: sdk/cognitivelanguage/azure-ai-language-questionanswering/azure/ai/language/questionanswering/projects/aio/_operations/_operations.py [hunk 25]
```diff
@@ -0,0 +1,1827 @@
...
+        :rtype: None
+        :raises: ~azure.core.exceptions.HttpResponseError
+
+        Example:
+            .. code-block:: python
+
+                # JSON input template you can fill out and use as your body input.
+                synonyms = {
+                    "nextLink": "str",  # Optional.
+                    "value": [
```
Comment (annatisch) [inline sdk/cognitivelanguage/azure-ai-language-questionanswering/azure/ai/language/questionanswering/projects/aio/_operations/_operations.py:1148]: Do we have a test/sample for `update_synonyms`?
This sample input block looks incorrect, it seems to be the output of the paging operation... presumably that's not the expected input structure?

Comment (mshaban-msft) [reply sdk/cognitivelanguage/azure-ai-language-questionanswering/azure/ai/language/questionanswering/projects/aio/_operations/_operations.py:1148]: this is the same request body as in the swagger [here](https://github.com/Azure/azure-rest-api-specs/blob/219b2b3dea0233e3cb43f72d2764c0e4e4d8fcd3/specification/cognitiveservices/data-plane/Language/stable/2021-10-01/questionanswering-authoring.json#L646)
and [here](https://github.com/Azure/azure-rest-api-specs/blob/219b2b3dea0233e3cb43f72d2764c0e4e4d8fcd3/specification/cognitiveservices/data-plane/Language/stable/2021-10-01/questionanswering-authoring.json#L1263)

**Review by annatisch — APPROVED** [review]

File: sdk/cognitivelanguage/azure-ai-language-questionanswering/azure/ai/language/questionanswering/projects/_version.py [hunk 26]
```diff
@@ -0,0 +1 @@
+from .._version.py import VERSION
```
Comment (annatisch) [inline sdk/cognitivelanguage/azure-ai-language-questionanswering/azure/ai/language/questionanswering/projects/_version.py:1]: Could you please update the underlying package version and add an update to the change log?

Comment (mshaban-msft) [reply sdk/cognitivelanguage/azure-ai-language-questionanswering/azure/ai/language/questionanswering/projects/_version.py:1]: what should be the version number?

Comment (annatisch) [reply sdk/cognitivelanguage/azure-ai-language-questionanswering/azure/ai/language/questionanswering/projects/_version.py:1]: I think in the chat we decided 1.1.0b1 :)

**Pull request merged**